client/webserver/site/src/js/markets.ts

```diff
@@ -1770,51 +1770,98 @@ export default class MarketsPage extends BasePage {
     const { page, market } = this
     let { baseUnitInfo, quoteUnitInfo, rateConversionFactor } = market
     const swapped = swap.estimate.value || 0
-    const fmtPct = percentFormatter.format
+    const fmtPct = (value: number) => {
+      if (value < 0.05) return '< 0.1'
+      return percentFormatter.format(value)
+    }
+
+    // If the asset is a token, in order to calculate the fee as a percentage
+    // of the total order, we try to  use the fiat rates to find out the
+    // exchange rate between the token and parent assets.
+    // Initially these are set to 1, which we would use if the asset is not a
+    // token and no conversion is needed.
+    let baseExchangeRate = 1
+    let quoteExchangeRate = 1
 
     if (market.base.token) {
       const parent = app().assets[market.base.token.parentID]
       baseUnitInfo = parent.unitInfo
+
+      const tokenFiatRate = app().fiatRatesMap[market.base.id]
+      const parentFiatRate = app().fiatRatesMap[parent.id]
+      if (tokenFiatRate && parentFiatRate) {
+        baseExchangeRate = tokenFiatRate / parentFiatRate
+      } else {
+        baseExchangeRate = 0
+      }
     }
+
+    console.log('baseExchangeRate', baseExchangeRate)
+
     if (market.quote.token) {
       const parent = app().assets[market.quote.token.parentID]
       quoteUnitInfo = parent.unitInfo
+
+      const tokenFiatRate = app().fiatRatesMap[market.quote.id]
+      const parentFiatRate = app().fiatRatesMap[parent.id]
+      if (tokenFiatRate && parentFiatRate) {
+        quoteExchangeRate = tokenFiatRate / parentFiatRate
+      } else {
+        quoteExchangeRate = 0
+      }
     }
 
-    let [toIsToken, fromIsToken] = [!!market.base.token, !!market.quote.token]
+    console.log('quoteExchangeRate', quoteExchangeRate)
+
     let [toUI, fromUI] = [baseUnitInfo, quoteUnitInfo]
+    let [toExchangeRate, fromExchangeRate] = [baseExchangeRate, quoteExchangeRate]
     if (this.currentOrder.sell) {
-      [fromIsToken, toIsToken] = [toIsToken, fromIsToken];
-      [fromUI, toUI] = [toUI, fromUI]
+      [fromUI, toUI] = [toUI, fromUI];
+      [fromExchangeRate, toExchangeRate] = [toExchangeRate, fromExchangeRate]
     }
 
+    console.log('fromExchangeRate', fromExchangeRate)
+    console.log('toExchangeRate', toExchangeRate)
+
     // Set swap fee estimates in the details pane.
-    const bestSwapPct = swap.estimate.realisticBestCase / swapped * 100
-    page.vSwapFeesLowPct.textContent = fromIsToken ? '' : `(${fmtPct(bestSwapPct)}%)`
+    const bestSwapPct = swap.estimate.realisticBestCase * fromExchangeRate / swapped * 100
```

Comment:
Hmm. This line seems wrong and the units don't pan out for a percent. Is this maybe the reason @JoeGruffins saw a crazy value on testnet.
![image](https://user-images.githubusercontent.com/6109680/217598949-44010e14-9cd5-4810-8cfd-22a810f49d03.png)

Reply:
Oooh. Nevermind. You're using `1` for non-tokens. The exchange rate actually needs to be applied to `swapped`, but the effect is the same. Carry on.